CReLU.lua

```diff
@@ -0,0 +1,51 @@
+local CReLU, parent = torch.class('nn.CReLU', 'nn.Sequential')
+
+-- Implements the CReLU activation function as described by
+-- W. Shang et al. in "Understanding and Improving Convolutional Neural Networks
+-- via Concatenated Rectified Linear Units"
+-- Note: Only supports batched input of the form [B,C,H,W]
+function CReLU:__init(inplace)
+   parent.__init(self)
+   local concatTable = nn.ConcatTable()
+   concatTable:add(nn.Identity())
+   concatTable:add(nn.MulConstant(-1))
+   self.inplace = inplace or false
+   self.batched = true
+   self:add(concatTable)
+   self:add(nn.JoinTable(2))
+   self:add(nn.ReLU(self.inplace))
+end
+
+
+function CReLU:updateOutput(input)
+   self.batched = true
+   if input:nDimension() == 3 then
+      self.batched = false
+   end
+
+   if self.batched then
+      parent.updateOutput(self, input)
+   else
+      parent.updateOutput(self, input:view(1, input:size(1), input:size(2), input:size(3)))
```

Comment:
ok, in that case, there's a lot of precedence for this. The constructor to nn.CReLU needs to take an argument nInputDims: https://github.com/torch/nn/search?utf8=%E2%9C%93&q=nInputDims

Then, updateOutput will become:
```lua
function CReLU:updateOutput(input)
   local input_
   local batched = input:dim() == self.nInputDims
   if not batched then
      input_ = input:view(1, -1)
  else
      input_ = input:view(input:size(1), -1)
  end
   parent.updateOutput(self, input)
   local osize = input:size()
   if not batched then
       osize[1] = osize[1] * 2
   else
      osize[2] = osize[2] * 2
   end
    self.output:resize(osize)
   return self.output
end
```

What do you think about this?